src/platform/webengine/BlinkWebViewProfileHelper.h

```diff
@@ -40,14 +43,22 @@ const char kWebSQL[] = "webSQL";
 
 class BlinkWebViewProfileHelper {
 public:
-    BlinkWebViewProfileHelper() {}
-    virtual ~BlinkWebViewProfileHelper() {}
+    static BlinkWebViewProfileHelper* instance();
 
     static void clearBrowsingData(const int removeBrowsingDataMask,
         webos::WebViewProfile* profile = nullptr);
     static void clearDefaultBrowsingData(const int removeBrowsingDataMask);
     static int maskForBrowsingDataType(const char* key);
-    static void setProxyRules(const std::string& proxy_rules);
+    webos::WebViewProfile* getProfile(const std::string& app_id);
+    void buildProfile(const std::string& app_id, const std::string& proxy_rules);
+
+private:
+    BlinkWebViewProfileHelper() {}
+    virtual ~BlinkWebViewProfileHelper() = default;
+
+    std::map<const std::string, webos::WebViewProfile*> app_profile_map_;
```

Comment:
Use Qt coding style in WAM (yes, eventually when Qt-less effort is done it will go away, but for now our coding standard is still matching Qt).

In this case, the variable should be named m_appProfileMap